packages/ts-morph/src/ProjectContext.ts

```diff
@@ -67,6 +68,8 @@ export class ProjectContext {
             this._customTypeChecker = new TypeChecker(this);
             this._customTypeChecker._reset(() => opts.typeChecker!);
         }
+        this.compilerFactory.onSourceFileAdded(() => this._projectVersion += 1);
+        this.compilerFactory.onSourceFileRemoved(() => this._projectVersion += 1);
```

Comment:
I'm looking at the TS source code and wow, you're right that this will really speed things up! Question though, does this not need to occur when a source file is changed? I might be missing something though.

Reply:
Ah yeah, I missed the call to reset from the ProjectContext.